tests/ui/suspicious_xor.stderr

```diff
@@ -0,0 +1,51 @@
+error: '^' is not the exponentiation operator
+  --> $DIR/suspicious_xor.rs:19:13
+   |
+LL |     let _ = 2 ^ 5;
+   |             ^^^^^ help: did you mean to write: `2.pow(5)`
+   |
+   = note: `-D clippy::suspicious-xor` implied by `-D warnings`
+
+error: '^' is not the exponentiation operator
+  --> $DIR/suspicious_xor.rs:20:13
+   |
+LL |     let _ = 2i32 ^ 9i32;
+   |             ^^^^^^^^^^^ help: did you mean to write: `2_i32.pow(9_i32)`
+
+error: '^' is not the exponentiation operator
+  --> $DIR/suspicious_xor.rs:21:13
+   |
+LL |     let _ = 2i32 ^ 2i32;
+   |             ^^^^^^^^^^^ help: did you mean to write: `2_i32.pow(2_i32)`
+
+error: '^' is not the exponentiation operator
+  --> $DIR/suspicious_xor.rs:22:13
+   |
+LL |     let _ = 50i32 ^ 3i32;
```

Comment:
I don't think this case should be warned. https://github.com/rust-lang/rust-clippy/pull/9506#pullrequestreview-1121045671

Reply:
I think 50 ^ 3 should be warned because, is really that common xoring 50 and 3? In my opinion the cases where this happens is as common as using a bitwise OR in an if statement. Both cases are just confusion with other operators in a vast majority of cases.

Reply:
It depends on contexts. If we warn any `x^y` (`x, y = literal integer`), the category should be `restriction` or `pedantic`.

Reply:
We’re only talking about linting literal using the decimal form which I’ve never seen used when knowingly xoring.

Message could suggest ‘if this is intentional, use the hexadecimal or binary form as this is more explicit’

Reply:
> Message could suggest ‘if this is intentional, use the hexadecimal or binary form as this is more explicit’
Are you suggesting to replace the current message (`did you mean to write: 50.pow(3)`) or to have both?